maven-release-manager/src/main/java/org/apache/maven/shared/release/phase/AbstractMapVersionsPhase.java

```diff
@@ -67,9 +75,16 @@
  * @author <a href="mailto:brett@apache.org">Brett Porter</a>
  * @author Robert Scholte
  */
+@Component( role = ReleasePhase.class, hint = "map-release-versions" )
 public abstract class AbstractMapVersionsPhase
```

Comment:
This is then inherited to the non-abstract class?

Reply:
Yes, it seems that way. If it wasn't then the scmRepositoryConfigurator would have remained null causing an NPE.

Reply:
I know the reason and I believe this is wrong. We have changed to constructor injection. Thus the implementing classes have `@Inject` annotated constructors and there your instance should be provided. @cstamas

Reply:
If I remove this like then I get an NPE as expected.

Reply:
Even with constructor injection? @cstamas Can you help? I have intentionally moved from Plexus annotations to standard Java annotations

Reply:
Ah, yes. I missed you mentioned @Inject.
Fixing it.

Reply:
Fixed.

Reply:
Fixed.